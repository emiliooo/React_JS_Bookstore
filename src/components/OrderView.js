import React ,{Component} from 'react';

export default class OrderView extends Component {

    render() {
        return (
            <div className = "orderView" >
                <span> { this.props.book.name} </span> 
                <i>{ this.props.book.author} </i>  <br/>
                <img src={ this.props.book.image } width="75" height="100" alt="" />
                <button className="btn btn-warning" onClick={ (event) => this.props.remove(this.props.book.name)}> Remove Order </button>
            </div>
        )
    }

}