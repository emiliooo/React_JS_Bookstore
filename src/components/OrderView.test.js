import React from 'react';

import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import OrderView from './OrderView'
configure({adapter : new Adapter()})

describe('Order view tests', () => {

    it('Title of the book is displayed as in props' , () => {
        const book = {
            name:"Book title"
        }
        const wrapper = shallow(<OrderView book = {book} />)
        expect(wrapper.find('.orderView').find('span').text()).toBe(book.name);
    })

})