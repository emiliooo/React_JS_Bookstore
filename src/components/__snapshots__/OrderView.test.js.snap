// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order view tests Snapshot matches 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrderView
    book={
      Object {
        "name": "Book title",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <b>
          Book title
        </b>,
        " ",
        <br />,
        <i>
           
        </i>,
        "  ",
        <br />,
        <img
          alt=""
          height="100"
          width="75"
        />,
        <button
          className="btn btn-warning"
          onClick={[Function]}
        >
           Remove Order 
        </button>,
      ],
      "className": "orderView",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Book title",
        },
        "ref": null,
        "rendered": "Book title",
        "type": "b",
      },
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          " ",
        ],
        "type": "i",
      },
      "  ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "height": "100",
          "src": undefined,
          "width": "75",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Remove Order ",
          "className": "btn btn-warning",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": " Remove Order ",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <b>
            Book title
          </b>,
          " ",
          <br />,
          <i>
             
          </i>,
          "  ",
          <br />,
          <img
            alt=""
            height="100"
            width="75"
          />,
          <button
            className="btn btn-warning"
            onClick={[Function]}
          >
             Remove Order 
          </button>,
        ],
        "className": "orderView",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Book title",
          },
          "ref": null,
          "rendered": "Book title",
          "type": "b",
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            " ",
          ],
          "type": "i",
        },
        "  ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "height": "100",
            "src": undefined,
            "width": "75",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Remove Order ",
            "className": "btn btn-warning",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": " Remove Order ",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Order view tests Title of the book is displayed as in props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrderView
    book={
      Object {
        "name": "Book title",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <b>
          Book title
        </b>,
        " ",
        <br />,
        <i>
           
        </i>,
        "  ",
        <br />,
        <img
          alt=""
          height="100"
          width="75"
        />,
        <button
          className="btn btn-warning"
          onClick={[Function]}
        >
           Remove Order 
        </button>,
      ],
      "className": "orderView",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Book title",
        },
        "ref": null,
        "rendered": "Book title",
        "type": "b",
      },
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          " ",
        ],
        "type": "i",
      },
      "  ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "height": "100",
          "src": undefined,
          "width": "75",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Remove Order ",
          "className": "btn btn-warning",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": " Remove Order ",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <b>
            Book title
          </b>,
          " ",
          <br />,
          <i>
             
          </i>,
          "  ",
          <br />,
          <img
            alt=""
            height="100"
            width="75"
          />,
          <button
            className="btn btn-warning"
            onClick={[Function]}
          >
             Remove Order 
          </button>,
        ],
        "className": "orderView",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Book title",
          },
          "ref": null,
          "rendered": "Book title",
          "type": "b",
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            " ",
          ],
          "type": "i",
        },
        "  ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "height": "100",
            "src": undefined,
            "width": "75",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Remove Order ",
            "className": "btn btn-warning",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": " Remove Order ",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
